sycl/source/detail/program_manager/program_manager.hpp

```diff
@@ -538,6 +547,10 @@ class ProgramManager {
   // their associated entry resources when they die.
   DeviceGlobalMap m_DeviceGlobals{/*OwnerControlledCleanup=*/true};
 
+  // Maps between host_pipe identifiers and associated kernel global
+  // information.
+  std::unordered_map<std::string_view, unsigned> m_KernelGlobalInfo;
```

Comment:
I would probably rename it something more specific to free function kernels.
```suggestion
  std::unordered_map<std::string_view, unsigned> m_FreeFunctionKernelGlobalInfo;
```
This should also be reflected in the names of functions referring to it.